libs/icu_headroom_metric.py

```diff
@@ -0,0 +1,168 @@
+from typing import Union, Optional
+import enum
+from covidactnow.datapublic.common_fields import CommonFields
+from libs.datasets import can_model_output_schema as schema
+
+
+import pandas as pd
+
+
+class CovidPatientsMethod(enum.Enum):
+    ACTUAL = 0
+    ESTIMATED = 1
+
+
+class NonCovidPatientsMethod(enum.Enum):
+    ACTUAL = 0
+    ESTIMATED_FROM_TYPICAL_UTILIZATION = 1
+    ESTIMATED_FROM_TOTAL_ICU_ACTUAL = 2
+
+
+DEFAULT_ICU_DECOMP = 0.21
+
+ICU_DECOMP_OVERRIDE = {
+    "AL": 0.15,
+    "AZ": 0.4,
+    "DE": 0.3,
+    "DC": 0.15,
+    "GA": 0.1,
+    # TODO(https://trello.com/c/1ddB5ntl/): CCM is currently giving us an
+    # extra-high utilization rate. If that gets fixed we may need to bump this
+    # back down.
+    "MS": 0.37,
+    "NV": 0.25,
+    "RI": 0,
+}
+
+
+def get_decomp_for_state(state: str) -> float:
+    return ICU_DECOMP_OVERRIDE.get(state, DEFAULT_ICU_DECOMP)
+
+
+class ICUMetricData:
+    def __init__(self, data, latest_values, decomp_factor):
+        self.data = data
+        self.latest_values = latest_values
+        self.decomp_factor = decomp_factor
+
+    @property
+    def actual_current_icu_covid(self) -> Optional[pd.Series]:
+        actuals = self.data[CommonFields.CURRENT_ICU]
+        if not actuals.any():
+            return None
+
+        return actuals
+
+    @property
+    def estimated_current_icu_covid(self) -> Optional[pd.Series]:
+        estimated = self.data[schema.CURRENT_ICU]
```

Comment:
FWIW `schema` seems overly generic to the point of being meaningless (what's the difference between `CommonFields.CURRENT_ICU` and `schema.CURRENT_ICU`? I have no idea.) If you need to shorten it, perhaps `model_schema`?